### PR TITLE
feat: aggregate metrics

### PR DESCRIPTION
**Pull request by jlegoff**
Up until now, each span generates a unique datapoint that is appended to the metrics. This PR keeps track of the datapoint and performs aggregations as new spans are processed.

### CODE REVIEW

File: connector/apmconnector/transaction.go [hunk 1]
```diff
@@ -298,14 +295,13 @@ func (transaction *Transaction) ProcessRootSpan() bool {
 		attributes.PutStr("transactionName", transactionName)
 		attributes.PutStr("metricTimesliceName", transactionName)
 
-		transaction.resourceMetrics.RecordHistogramFromSpan("apm.service.transaction.duration", attributes, span)
+		transaction.resourceMetrics.AddHistogramFromSpan("apm.service.transaction.duration", attributes, span)
 
 		if remainingNanos > 0 {
+			// FIXME this is already in the map
```
Comment (sdaubin) [inline connector/apmconnector/transaction.go:301]: 🤔 Should we just remove the `PutStr` invocation with `transactionName`?

**Review by sdaubin — APPROVED** [review]
Works for me locally.  Thanks!

**Review by alanwest — APPROVED** [review]
yup, works for me!

**Review by rubenruizdegauna — APPROVED** [review]
LGTM

**Pull request merged**